__tests__/tests.ts

```diff
@@ -585,6 +589,143 @@ describe('htmlbars-inline-precompile', function () {
+    `);
+      expect(transformed).toContain(`"scope": () => [two]`);
+    });
+
+    it('allows AST transform to bind a JS import', function () {
+      precompile = runASTTransform(compiler, function (env) {
+        return {
+          name: 'sample-transform',
+          visitor: {
+            PathExpression(node) {
+              if (node.original === 'onePlusOne') {
+                let name = env.meta.jsutils.bindImport('my-library', 'default', {
+                  nameHint: 'two',
+                });
+                return env.syntax.builders.path(name);
+              }
+              return undefined;
+            },
+          },
+        };
+      });
+
+      let transformed = transform(stripIndent`
+        import { precompileTemplate } from '@ember/template-compilation';
+        const template = precompileTemplate('<Message @text={{onePlusOne}} />');
+      `);
+
+      expect(transformed).toContain(`@text={{two}}`);
+      expect(transformed).toContain(`locals: [two]`);
+      expect(transformed).toContain(`import two from "my-library"`);
+    });
+
+    it('does not smash existing binding for import', function () {
+      precompile = runASTTransform(compiler, function (env) {
+        return {
+          name: 'sample-transform',
+          visitor: {
+            PathExpression(node) {
+              if (node.original === 'onePlusOne') {
+                let name = env.meta.jsutils.bindImport('my-library', 'default', {
+                  nameHint: 'two',
+                });
+                return env.syntax.builders.path(name);
+              }
+              return undefined;
+            },
+          },
+        };
+      });
+
+      let transformed = transform(stripIndent`
+        import { precompileTemplate } from '@ember/template-compilation';
+        export function inner() {
+          let two = 'twice';
+          const template = precompileTemplate('<Message @text={{onePlusOne}} />');
+        }
+      `);
+
+      expect(transformed).toContain(`@text={{two0}}`);
+      expect(transformed).toContain(`locals: [two0]`);
+      expect(transformed).toContain(`import two0 from "my-library"`);
+    });
+
+    it('does not smash existing binding for expression', function () {
+      precompile = runASTTransform(compiler, function (env) {
+        return {
+          name: 'sample-transform',
+          visitor: {
+            PathExpression(node) {
+              if (node.original === 'onePlusOne') {
+                let name = env.meta.jsutils.bindValue('1+1', { nameHint: 'two' });
+                return env.syntax.builders.path(name);
+              }
+              return undefined;
+            },
+          },
+        };
+      });
+
+      let transformed = transform(stripIndent`
+        import { precompileTemplate } from '@ember/template-compilation';
+        export default function() {
+          let two = 'twice';
+          const template = precompileTemplate('<Message @text={{onePlusOne}} />');
```

Comment:
What about bindings within the template itself?

```hbs
{{#let "two" as |two|}}
  <Message @text={{onePlusOne}} />
{{/let}}
```

Reply:
Good idea, we need to avoid the intersection of both the js and hbs existing bindings.